nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,143 @@
+                - One with the `model_class.objects.first().pk`
+                - One with the `model_class.objects.last().pk`
+            - Any model list view URL pattern will have two endpoints:
+                - One with default pagination
+                - One with custom pagination (5 pages with <total_object_count//5> instances per page)
+            - Any generic endpoint like `core:home` will have one endpoint which is the URL pattern itself.
+        """
+        # If the model class is found, then we know we are dealing with a model related endpoint
+        if model_class:
+            if len(model_class.objects.all()) > 1:
+                # Handle detail view url patterns
+                if "_list" not in view_name:
+                    # Identify the placeholder for the uuid
+                    replace_string = ""
+                    if "<uuid:pk>" in url_pattern:
+                        replace_string = "<uuid:pk>"
+                    elif "(?P<pk>[/.]+)" in url_pattern:
+                        replace_string = "(?P<pk>[/.]+)"
+
+                    if replace_string:
+                        # Replace the uuid with the actual uuid
+                        first_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.first().pk))
+                        second_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.last().pk))
+                        if view_name not in self.app_name_to_urls["endpoints"]:
+                            self.app_name_to_urls["endpoints"][view_name] = []
+                        self.app_name_to_urls["endpoints"][view_name].append(first_url_pattern)
+                        self.app_name_to_urls["endpoints"][view_name].append(second_url_pattern)
+                # Handle list view url patterns
+                else:
+                    if view_name not in self.app_name_to_urls["endpoints"]:
+                        self.app_name_to_urls["endpoints"][view_name] = []
+                    # One endpoint with default pagination
+                    self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+                    total_count = len(model_class.objects.all())
+                    page_query_parameter = 5
+                    per_page_query_parameter = total_count // page_query_parameter
+                    # One endpoint with default pagination
+                    self.app_name_to_urls["endpoints"][view_name].append(
+                        url_pattern
+                        + "?per_page="
+                        + str(per_page_query_parameter)
+                        + "&"
+                        + "page="
+                        + str(page_query_parameter)
+                    )
+            else:
+                # TODO handle the case where there is no instances of the model is found
+                self.std.out.write(f"No instances of {model_class} found")
```

Comment:
Currently skipping over model classes that we do not have at least 2 instances of.